polyfill/lib/plaindate.mjs

```diff
@@ -34,77 +35,77 @@ export class PlainDate {
   get era() {
     if (!ES.IsTemporalDate(this)) throw new TypeErrorCtor('invalid receiver');
     const isoDate = ES.TemporalObjectToISODateRecord(this);
-    return ES.CalendarEra(GetSlot(this, CALENDAR), isoDate);
+    return calendarImpl(GetSlot(this, CALENDAR)).isoToDate(isoDate, { era: true }).era;
```

Comment:
`calendarImpl(GetSlot(this, CALENDAR))` is repeated 56 times in this PR. Maybe replace it with an exported function like `getCalendarImpl(this)` (from calendar.mjs) or a `CalendarImpl(this)` export from ecmascript.mjs?